src/x/context/finalizeable_list_gen.go

```diff
@@ -129,7 +136,22 @@ func (l *finalizeableList) Init() *finalizeableList {
 	l.root.prev = &l.root
 	l.len = 0
 	if l.Pool == nil {
-		l.Pool = newFinalizeableElementPool(nil)
+		// Use a static pool at least, otherwise each time
+		// we create a list with no pool we create a wholly
+		// new pool of finalizeables (4096 of them).
+		defaultFinalizeableElementsPoolLock.RLock()
+		l.Pool = defaultFinalizeableElementsPool
+		defaultFinalizeableElementsPoolLock.RUnlock()
+		if l.Pool == nil {
+			defaultFinalizeableElementsPoolLock.Lock()
+			if defaultFinalizeableElementsPool == nil {
+				// Still not set, allocate pool once.
+				defaultFinalizeableElementsPool = newFinalizeableElementPool(nil)
+			}
+			// Take ref to guaranteed allocated pool.
+			l.Pool = defaultFinalizeableElementsPool
+			defaultFinalizeableElementsPoolLock.Unlock()
+		}
```

Comment:
Would it be worth using `sync.Once` instead? Something like

```golang
var elementPoolInit sync.Once
// ...
elementPoolInit.Do(func(){
  defaultFinalizeableElementsPool = newFinalizeableElementPool(nil)
})
```

Reply:
Good point, will do.